universal-login-commons/lib/core/utils/messages/computeGasData.ts

```diff
@@ -0,0 +1,22 @@
+import {ensure} from '../errors';
+import {isProperHexString} from '../hexStrings';
+
+export const ZERO_BYTE_GAS_COST = 4;
+export const NON_ZERO_BYTE_GAS_COST = 68;
+
+export const computeGasData = (data: string) => {
+  ensure(isProperHexString(data), Error, 'Not a valid hex string');
+
+  if (data === '0x') {
+    return 0;
+  }
+
+  return data
+    .slice(2)
+    .match(/.{2}/g)!
+    .reduce((totalGasCost, byte) =>  totalGasCost + gasCostFor(byte), 0);
```

Comment:
btw if you reverse match with slice you will not need extra if at the begining.
So with:
```ts
  return data
    .match(/.{2}/g)
    .slice(1)
    .reduce((totalGasCost, byte) =>  totalGasCost + gasCostFor(byte), 0);
```
you will not need preceding:
```ts
  if (data === '0x') {
    return 0;
  }
```